### PR TITLE
Make BA code optional

### PR DESCRIPTION
**Pull request by andrewcroce**
When all other fields fail... this field stands alone

screen shot 2018-09-25 at 2 34 22 pm

### CODE REVIEW

File: atst/forms/financial.py [hunk 1]
```diff
@@ -121,9 +121,9 @@ def is_missing_task_order_number(self):
     )
 
     ba_code = StringField(
-        "Program Budget Activity (BA) Code",
+        "Program Budget Activity (BA) Code (Optional)",
         description="BA Code is used to identify the purposes, projects, or types of activities financed by the appropriation fund. <br/><em>It should be two digits, followed by a letter.</em>",
```
Comment (patricksmithdds) [inline atst/forms/financial.py]: While we're here, can we update the text here to indicate the letter is optional as well.

**Comment by montana-mil** [comment]
I ended up just taking the Optional() validator out altogether. The form works as expected and the tests pass. When `Optional()` was included (in either argument position), there were about 7 tests not passing inexplicably. @dandds and I went over this and agreed this is the best way to handle it, though it feels uncomfortable since we don't know why it works. (Or rather, why including `Optional()` *didn't* work)

**Comment by patricksmithdds** [comment]
@montana-mil @dandds I poked a little bit more into why `Optional()` wasn't working and I think it's just a problem with the tests. The `Optional` validator [checks whether `raw_data` is set](https://github.com/wtforms/wtforms/blob/283b2803206825158834f1828bbf749c129b7c47/src/wtforms/validators.py#L213-L240), and the field's `raw_data` is only set [when the field receives the `form_data` kwarg](https://github.com/wtforms/wtforms/blob/283b2803206825158834f1828bbf749c129b7c47/src/wtforms/fields/core.py#L335) (which is [passed in from the form](https://github.com/wtforms/wtforms/blob/283b2803206825158834f1828bbf749c129b7c47/src/wtforms/form.py#L110-L116)).

In the routes, `flask_wtf` handles setting the `form_data` for us, but in tests where we operating outside of a request context, we have to do it manually. 

All of that is to say, we can use the `Optional()` validator if we update the tests like:

```diff
diff --git a/tests/forms/test_financial.py b/tests/forms/test_financial.py
index 970e4b6b..dc89ab30 100644
--- a/tests/forms/test_financial.py
+++ b/tests/forms/test_financial.py
@@ -74,8 +74,8 @@ def test_treasury_code_validation(input_, expected):
     ],
 )
 def test_ba_code_validation(input_, expected):
-    form_data = {"ba_code": input_}
-    form = FinancialForm(data=form_data)
+    form_data = ImmutableMultiDict([("ba_code", input_)])
+    form = FinancialForm(form_data)
     form.validate()
     is_valid = "ba_code" not in form.errors
```

**Comment by andrewdds** [comment]
I misunderstood the request. The field is NOT optional, just the letter in the code is optional.
`Required()` validator added back, just changed the description.

File: atst/forms/financial.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import pendulum
 from wtforms.fields.html5 import DateField, EmailField
 from wtforms.fields import StringField, FileField
-from wtforms.validators import InputRequired, Required, Email, Regexp
+from wtforms.validators import InputRequired, Required, Optional, Email, Regexp
```
Comment (patricksmithdds) [inline atst/forms/financial.py]: This `Optional` import isn't used anywhere anymore.

**Pull request merged**